olive/data/container/data_container.py

```diff
@@ -38,6 +39,13 @@ def load_dataset(self, data_root_path: Optional[str] = None):
         data_dir = get_local_path_from_root(data_root_path, data_dir)
         if data_dir:
             params_config["data_dir"] = data_dir
+
+        if params_config.get("token", False):
+            from huggingface_hub import login
+
+            token = get_huggingface_token()
```

Comment:
is this manual login needed? I think just passing the value of `token` (either a boolean or the token string) to the `load_dataset` should work? 
https://huggingface.co/docs/datasets/v2.14.7/en/package_reference/loading_methods#datasets.load_dataset.token

https://github.com/microsoft/Olive/blob/main/olive/data/component/load_dataset.py#L35
https://github.com/microsoft/Olive/blob/main/examples/open_llama/lora_user_script.py#L14

There is no need for us to do the login.

Reply:
I think where you are getting a login requirement might be from the preprocessing component when it loads the tokenizer. 
In that case, we either need to expose `token` argument to the component or do a more centralized login. not here in the load dataset part.

Reply:
I personally prefer the "expose token" solution so that we don't need to login manually. The manual login assumes things about the location of the token but it might be stored in other places based on the transformers related environment variables or if they used git credentials.
https://huggingface.co/docs/huggingface_hub/package_reference/environment_variables#hfhome

Reply:
I was also thinking about these parameters for a while but didn't get the time to work on it. 

Just opened this PR https://github.com/microsoft/Olive/pull/728
Please let me know if it fulfills your requirements for login. For AzureML and Docker systems, we will have to think about it a bit more.

Reply:
I'm okay with your change for local system, but we still need str token for aml system. I'll double check with security team about this.